### PR TITLE
fix: handle Auto delimiter in fromDecimal (#2217)

### LINKED ISSUE #2217: Bug report: fromDecimal(...) function does not correctly handle delim parameter passed as default ("Auto")
**Describe the bug**
When fromDecimal is called without specifying a value for the "delim" parameter, or when delim is passed as "Auto" it will only parse the first number in the passed text ignoring any further numbers whatever separator has been used.

**To Reproduce**
Steps to reproduce the behaviour or a link to the recipe / input used to cause the bug:

See bug report #2139 which arises as a consequence of this issue.

**Expected behaviour**
delim = "Auto" case should be handled correctly, in a manner similar to the fromHex(...) function.

**Additional context**
fromHex could be used as a guide towards a correct implementation, noting that there are other differences because pairs of hex digits can be packed exactly into bytes whereas for decimal up to 3 digits can be required.

### PR DESCRIPTION
**Pull request by OndraDol**
## Summary

Fixes the fromDecimal() function to correctly handle the Auto delimiter mode. Utils.charRep("Auto") returns undefined, causing data.split(undefined) to treat the entire input as a single element and only parse the first decimal value.

## Changes

- Use regex `/[^\d-]+/` to split on any non-digit, non-minus character when delimiter is "Auto", consistent with how fromHex() handles its Auto mode
- Added test cases for Auto delimiter with comma, colon, and semicolon separators

## Related Issue

Closes #2217

## Testing

- Verified fix with space, comma, colon, and semicolon separators
- Verified signed value support still works
- Verified explicit delimiter mode is unaffected
- ESLint passes clean

### CODE REVIEW

File: tests/node/tests/operations.mjs [hunk 1]
```diff
@@ -532,6 +532,12 @@ Top Drawer`, {
         assert.strictEqual(chef.fromDecimal("72 101 108 108 111").toString(), "Hello");
     }),
 
+    it("From decimal with Auto delimiter", () => {
+        assert.strictEqual(chef.fromDecimal("72,101,108,108,111").toString(), "Hello");
+        assert.strictEqual(chef.fromDecimal("72:101:108:108:111").toString(), "Hello");
+        assert.strictEqual(chef.fromDecimal("72;101;108;108;111").toString(), "Hello");
+    }),
```
Comment (C85297) [inline tests/node/tests/operations.mjs]: Would you mind adding a test to check the functionality of handling negative numbers? From reading your code, I believe that the output of `chef.fromDecimal("-72 101 108 -108 -111")` would be `[-72, 101, 108, -108, -111]` and it would be good to confirm whether that is or is not the case in a test.

**Review by C85297 — CHANGES_REQUESTED** [review]

**Comment by C85297** [comment]
Closing until CLA is signed
